pkg/persistence/persistence.go

```diff
@@ -86,6 +90,25 @@ func CreateLocationKeyStore(keystoreUrl string) (LocationKeyStore, error) {
 
 func InitLocationKeyStore() error {
 	var err error
-	keystore, err = CreateLocationKeyStore(pkg.Config.KeystoreUrl)
+	keystoreUrl := pkg.Config.KeystoreUrl
+
+	// Check if MongoDB should be used instead of KeystoreUrl
+	log.Debug("Checking if we should use MongoDB")
+	if pkg.Config.MongodbServer != "" {
+		if pkg.Config.MongodbUsername != "" {
+			log.Debug("MongodbServer and MongodbUsername provided, using MongoDB")
+			keystoreUrl = fmt.Sprintf(
```

Comment:
going back on what I said earlier.
Can We keep a copy of the keystore URL with ONLY the port and host?  
mongo://hiddenuser:hiddenpassword@host:port (or what ever it is)
Log the user /password obscured URL in log levels < TRACE and show the full one only in trace?